mlir/lib/Dialect/NVGPU/IR/NVGPUDialect.cpp

```diff
@@ -375,15 +375,9 @@ LogicalResult WarpgroupGenerateDescriptorOp::verify() {
   MemRefType memrefType = getTensor().getType();
   MemRefType tensorMapType = getTensorMap().getType().getTensor();
 
-  if (memrefType != tensorMapType)
-    return emitError() << "memref and tensor map type mismatch";
```

Comment:
What's the semantic of unmatched memref/tensor types?

Could you add a test case demonstrating this case?

Reply:
The issue is similar what we discuss [here](https://github.com/llvm/llvm-project/pull/69584#issuecomment-1777594817).